cwl/src/main/scala/wdl4s/cwl/CwlFile.scala

```diff
@@ -33,28 +33,28 @@ import wdl4s.wom.{CommandPart, RuntimeAttributes}
   * @param temporaryFailCodes
   * @param permanentFailCodes
   */
-case class CommandLineTool(
-                            inputs: Array[CommandInputParameter] = Array.empty,
-                            outputs: Array[CommandOutputParameter] = Array.empty,
-                            `class`: Witness.`"CommandLineTool"`.T = "CommandLineTool".narrow,
-                            id: Option[String] = None,
-                            requirements: Option[Array[Requirement]] = None,
-
-                            //TODO: Fix this when CwlAny parses correctly
-                            //hints: Option[Array[CwlAny]] = None,
-                            hints: Option[Array[Map[String, String]]] = None,
-
-                            label: Option[String] = None,
-                            doc: Option[String] = None,
-                            cwlVersion: Option[CwlVersion] = Option(CwlVersion.Version1),
-                            baseCommand: Option[BaseCommand] = None,
-                            arguments: Option[Array[CommandLineTool.Argument]] = None,
-                            stdin: Option[StringOrExpression] = None,
-                            stderr: Option[StringOrExpression] = None,
-                            stdout: Option[StringOrExpression] = None,
-                            successCodes: Option[Array[Int]] = None,
-                            temporaryFailCodes: Option[Array[Int]] = None,
-                            permanentFailCodes: Option[Array[Int]] = None) {
+case class CommandLineTool private(
```

Comment:
So circe is using this `private` constructor somehow... but from scala we can use the `apply`?

Reply:
yes